arvo/clay/data-types.md

```diff
@@ -23,150 +24,180 @@ at least skimming, this so that you get a rough idea of how our state is
+    +$  rove                                                ::  stored request
+          $%  [%sing =mood]                             ::  single request
+              [%next =mood aeon=(unit aeon) =cach]      ::  next version of one
+              $:  %mult                                 ::  next version of any
+                  =mool                                 ::  original request
+                  aeon=(unit aeon)                      ::  checking for change
+                  old-cach=(map [=care =path] cach)     ::  old version
+                  new-cach=(map [=care =path] cach)     ::  new version
+              ==                                        ::
+              [%many track=? =moat lobes=(map path lobe)] ::  change range
+          ==                                            ::
 ```
 
-This represents a request for the state of the desk at a particular
-commit, specfied by `q`. `p` specifies what kind of information is
-desired, and `r` specifies the path we are requesting.
+Like a `rave` but with caches of current versions for `%next` and `%many`.
+Generally used when we store a request in our state somewhere. This is so that
+we can determine whether new versions actually affect the path we're subscribed to.
 
-#### `++moat:clay`, range subscription request
+
+#### `$mood:clay`, single subscription request
 
 ```hoon
-    ++  moat  ,[p=case q=case r=path]                       ::  change range
+  +$  mood  [=care =case =path]                         ::  request in desk
 ```
 
-This represents a request for all changes between `p` and `q` on path
-`r`. You will be notified when a change is made to the node referenced
-by the path or to any of its children.
+This represents a request for the state of the `desk` at a particular
+commit, specfied by `case`. `care` specifies what kind of information is
+desired, and `path` specifies the path we are requesting.
 
-### `++moot`, stored range subscription request
+#### `$moat:clay`, range subscription request
 
 ```hoon
-    ++  moot  ,[p=case q=case r=path s=(map path lobe)]     ::
+  +$  moat  [from=case to=case =path]                   ::  change range
 ```
 
-This is just a `++moat:clay` plus a map of paths to lobes. This map
-represents the data at the node referenced by the path at case `p`, if
-we've gotten to that case (else null). We only send a notification along
-the subscription if the data at a new revision is different than it was.
+This represents a request for all changes between `from` and `to` on `path`. You
+will be notified when a change is made to the node referenced by the `path` or to
+any of its children.
 
-### `++care:clay`, clay submode
+### `$care:clay`, Clay submode
 
 ```hoon
-    ++  care  ?(%u %v %w %x %y %z)                          ::  clay submode
+  +$  care  ?(%a %b %c %d %e %f %p %r %s %t %u %v %w %x %y %z)  ::  clay submode
 ```
 
+TODO: The rest of these. Ask Ted if there is any docs on them because a lot of
```

Comment:
I don't think there are any docs on them, but if there are some, they'd probably be in the Clay comments.

Reply:
Btw, generally @philipcmonk knows more about Clay internals than I do, except for Ford, which I might know a bit more intimately.

Reply:
A partial listing:
%a build hoon file at path
%b build mark by name (dynamically typed, i.e. a $dais datastructure)
%c build mark conversion gate by "from" and "to" mark names, dynamically typed (a $tube)
%d build mark, statically typed (vase of type-parameterized $nave core)
%e build mark conversion gate, statically typed
...
now from the end
...
%z hash of path contents (file, directory, or both)
%y directory listing
%x file contents

I don't recall the others offhand.

Reply:
Actually `%d` lists your desks, `%e` gives the nave, `%f` gives the statically typed mark conversion gate

`%u` now just checks whether the file exists, not what this page says.
`%p` produces the permissions for a directory
`%r` is like `%x` but produces the result as a vase instead of statically
`%s` has miscellaneous debug endpoints
`%t` produce list of descendant paths for a directory
`%v` get whole `dome` for a desk.  When used on a foreign desk this gets us "up-to-date" to the requested version.
`%w` get the version number and date of the requested path.  For example, `.^([@ud @da] %cw /=home`)` gives you the current version number and last revision date for the `%home` desk

Reply:
How's this? I feel like this part basically needs its own separate tutorial since as its written its not apparent how to immediately put this stuff to use, but I think that's outside of the scope of this doc. Maybe something for @tinnus-napbus to look into

Reply:
https://github.com/urbit/docs/pull/1055/commits/c45d8c857a805b4895800f9dc76ea7ccc10cd4f1

Reply:
I've documented usage in a clay scry reference (see [https://github.com/tinnus-napbus/clay-docs/blob/main/scry.md](https://github.com/tinnus-napbus/clay-docs/blob/main/scry.md)) so these short summaries will be sufficient in the data types reference I think

Reply:
Awesome, updated my descriptions a bit further. Will need to link to your page on clay scries once its done, but this makes me happy with the concise descriptions I have here.